btsmarthub_devicelist/__init__.py

```diff
@@ -0,0 +1,342 @@
+import requests
+from urllib.parse import quote
+import json
+import math
+import random
+import hashlib
+import logging
+import re
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def compute_md5hash(hashstring):
+    """Return the MD5 Hash of a given string"""
+    m = hashlib.md5()
+    m.update(hashstring.encode('utf-8'))
+    return m.hexdigest()
+
+
+def update_label(source, labels):
+    # sort so we do larges matches first removing issues on things like 'ip' used in longer strings
+    labels.sort(key=len, reverse=True)
+    for label in labels:
+        source = source.replace(label + ":", "\"" + label + "\":")
+
+    return source
+
+
+DEFAULT_IP = '192.168.1.254'
+
+
+class BTSmartHub(object):
+    """ Represents a session to a BT SmartHub Router."""
+
+    def __init__(self, router_ip=None, smarthub_model=None):
+        """ Initialize the router and if no router model is passed, try to determine model"""
+
+        if router_ip:
+            self.router_ip = router_ip
+        else:
+            self.router_ip = DEFAULT_IP
+
+        if not smarthub_model:
+            self.smarthub_model = self.autodetect_smarthub_model()
+        else:
+            self.smarthub_model = smarthub_model
+
+    def get_devicelist(self, only_active_devices=None):
+
+        if only_active_devices is None:
+            only_active_devices = True
+        elif only_active_devices:
+            only_active_devices = True
+        else:
+            only_active_devices = False
```

Comment:
To ensure it is a boolean, you could replace this if/elif/else with `only_active_devices = bool(only_active_devices)`